tempodb/tempodb.go

```diff
@@ -243,17 +248,41 @@ func (rw *readerWriter) Find(ctx context.Context, tenantID string, id encoding.I
 	span, derivedCtx := opentracing.StartSpanFromContext(ctx, "store.Find")
 	defer span.Finish()
 
+	blockStartUUID, err := uuid.Parse(blockStart)
+	if err != nil {
+		return nil, metrics, err
+	}
+	blockStartBytes, err := blockStartUUID.MarshalBinary()
+	if err != nil {
+		return nil, metrics, err
+	}
+	blockEndUUID, err := uuid.Parse(blockEnd)
+	if err != nil {
+		return nil, metrics, err
+	}
+	blockEndBytes, err := blockEndUUID.MarshalBinary()
+	if err != nil {
+		return nil, metrics, err
+	}
+
 	rw.blockListsMtx.Lock()
 	blocklist, found := rw.blockLists[tenantID]
 	copiedBlocklist := make([]interface{}, 0, len(blocklist))
+
 	for _, b := range blocklist {
 		// if in range copy
 		if bytes.Compare(id, b.MinID) != -1 && bytes.Compare(id, b.MaxID) != 1 {
-			copiedBlocklist = append(copiedBlocklist, b)
+			blockIDBytes, _ := b.BlockID.MarshalBinary()
+			// check block is in shard boundaries
+			// blockStartBytes <= blockIDBytes < blockEndBytes
+			if (bytes.Compare(blockIDBytes, blockStartBytes) >= 0) && (bytes.Compare(blockIDBytes, blockEndBytes) < 0) {
```

Comment:
this needs to be inclusive on both sides.  otherwise you won't ever search a block with the guid BlockIDMax.